python/dgl/contrib/dis_kvstore.py

```diff
@@ -11,179 +11,106 @@
-        server_addr=server_namebook[server_id],
-        num_client=num_client,
-        msg_queue_size=msg_queue_size)
-
-    for name, data in ndata.items():
-        server.init_data(name=name, data_tensor=data)
-
-    for name, data in edata.items():
-        server.init_data(name=name, data_tensor=data)
-
-    if ndata_g2l is not None:
-        for name, data in ndata_g2l.items():
-            server.set_global2local(name=name, global2local=data)
-
-    if edata_g2l is not None:
-        for name, data in edata_g2l.items():
-            server.set_global2local(name=name, global2local=data)
-
-    print("start server %d on %s" % (server.get_id(), server.get_addr()))
-
-    server.start()
-
-
-def start_client(ip_config, ndata_partition_book, edata_partition_book, close_shared_mem=False, msg_queue_size=2*1024*1024*1024):
-    """Start a kvclient node.
-
-    Parameters
-    ----------
-    ip_config : str
-        Filename of server IP configure file.
-    ndata_partition_book : dict of tensor (mx.ndarray or torch.tensor)
-        Data mapping of node ID to server ID
-    edata_partition_book : dict of tensor (mx.ndarray or torch.tensor)
-        Data mapping of edge ID to server ID
-    close_shared_mem : bool
-        Close local shared-memory tensor access.
-    msg_queue_size : int
-        Size of message queue (2GB by default)
-
-    Returns
-    -------
-    KVClient
-        client handle
-    """
-    assert len(ip_config) > 0, 'ip_config cannot be empty.'
-    assert len(ndata_partition_book) > 0, 'ndata_partition_book cannot be empty.'
-    assert len(edata_partition_book) > 0, 'edata_partition_book cannot be empty.'
-
-    server_namebook = read_ip_config(ip_config)
-
-    client = KVClient(server_namebook=server_namebook, close_shared_mem=close_shared_mem, msg_queue_size=msg_queue_size)
-
-    for name, data in ndata_partition_book.items():
-        client.set_partition_book(name=name, partition_book=data)
-
-    for name, data in edata_partition_book.items():
-        client.set_partition_book(name=name, partition_book=data)
-
-    client.connect()
-
-    print("Client %d (%s) connected to kvstore ..." % (client.get_id(), client.get_addr()))
-
-    return client
-    
-
 class KVServer(object):
     """KVServer is a lightweight key-value store service for DGL distributed training.
 
-    In practice, developers can use KVServer to hold large graph features or graph embeddings 
-    across machines in a distributed setting. User can re-wriite _push_handler and _pull_handler 
-    to support flexibale algorithms.
+    In practice, developers can use KVServer to hold large-scale graph features or 
+    graph embeddings across machines in a distributed setting. User can re-wriite _push_handler() 
+    and _pull_handler() API to support flexibale algorithms.
+
+    DGL kvstore supports backup servers. That means we can lunach many servers on the same machine and all of 
+    these servers share the same shared-memory tensor.
```

Comment:
Sure, I will change it.

Reply:
Does the backup servers act like replicas? All can access the data and update the data. 
How to avoid write conflicts? Or in async scenarios, we don't care w/r, w/w conflicts in kv strore?